src/storage/compression/rle.cpp

```diff
@@ -132,8 +132,7 @@ struct RLECompressState : public CompressionState {
 	static idx_t MaxRLECount() {
 		auto entry_size = sizeof(T) + sizeof(rle_count_t);
 		auto entry_count = (Storage::BLOCK_SIZE - RLEConstants::RLE_HEADER_SIZE) / entry_size;
-		auto max_vector_count = entry_count / STANDARD_VECTOR_SIZE;
-		return max_vector_count * STANDARD_VECTOR_SIZE;
+		return entry_count;
```

Comment:
Yes, the old behavior would align this to the vector size. Switching between block sizes here could introduce a magic number. I.e., which block size is 'small'. Also, this only affects the last vector, which might be partially filled, no? So for bigger block sizes, we already alleviate the impact by having mostly full vectors?

Reply:
Now that I think about it, this might actually be a thing, because it affects the last vector, and then all segments after it I believe. So for example consider we have a column with 2 segments:

ColumnData:
- Segment 1: RLE
- Segment 2: Bitpacking

Since RLE is now no longer vector aligned at the end, we will start scanning segment 2 with a partial scan. Then the whole scan of Bitpacking would be no longer vector aligned. For bitpacking at least, this will result in a slightly less efficient scan as that prefers to be vector aligned.

I'm not 100% sure on this, but this might justify a deeper look. In general I think it would be valuable to know how important it is to vector align the segments. ( and perhaps it would also be nice to have some tests that force vector-unaligned scans to ensure that there are no hidden bugs there )

Reply:
I don't think that scenario is possible, compression is selected on a per-row-group basis not on a per-segment basis. For RLE there is not much of a difference in terms of performance whether or not segments are vector-aligned so it shouldn't matter too much. For bitpacking I suppose we already/still do vector-alignment across segments?

Reply:
@Mytherin ah right that's true! that is indeed not a problem then